R/src/feather-types.cpp

```diff
@@ -226,7 +230,7 @@ SEXP toSEXP(const ColumnPtr& x) {
   case feather::ColumnType::PRIMITIVE:
     return toSEXP(val);
   case feather::ColumnType::CATEGORY: {
-    IntegerVector out = Rf_allocVector(INTSXP, val->length);
```

Comment:
You can also use e.g. `IntegerVector out = no_init(val->length);` if all you want is an R vector with allocated but not zeroed memory.